### PR TITLE
Creation of Notes

### PR DESCRIPTION
**Pull request by moos3**
I have added the ability to create notes for an incident. This added support for the functionality for note creation per the PagerDuty API:
https://developer.pagerduty.com/documentation/rest/incidents/notes/create

### CODE REVIEW

File: lib/pagerduty/pagerduty.rb [hunk 1]
```diff
@@ -535,6 +535,37 @@ def notes(id)
...
+  #     * 'role'<~String>
+  #     * 'avatar_url'<~String>
+  #     * 'user_url'<~String>
+  #     * 'invitation_sent'<~Boolean>
+  #     * 'marketing_opt_out'<~Boolean>
+  #   * 'content'<~String>
+  #   * 'created_at'<~String>
+  # {Pagerduty API Reference}[https://developer.pagerduty.com/documentation/rest/schedules/overrides/create]
+  def create_note(options={})
+    Pagerduty::Schedules::Overrides::Override.new(curl({
```
Comment (kryptek) [inline lib/pagerduty/pagerduty.rb]: Why are you returning this object type? The response from the API should be an Array of Note objects. (See: https://developer.pagerduty.com/documentation/rest/incidents/notes/create)

File: lib/pagerduty/pagerduty.rb [hunk 2]
```diff
@@ -535,6 +535,37 @@ def notes(id)
...
+  # Create an note for a specific incident.
+  #
+  # ==== Parameters
+  # * 'options'<~Hash>
+  #   * 'id' - the id of the incident
+  #   * 'requester_id'<~String>
+  #   * 'note' - <~Hash>
+  #      * 'content' - <~ String>
+  # ==== Returns
+  # * <~Pagerduty::Schedules::Overrides::Override>
```
Comment (kryptek) [inline lib/pagerduty/pagerduty.rb]: Fix the documentation to reflect what this function actually returns

**Comment by kryptek** [comment]
Thanks for the PR. Just needs some slight modifications and I'll merge it in.

**Comment by moos3** [comment]
Kryptek should be good to go now.

File: lib/pagerduty/pagerduty.rb [hunk 3]
```diff
@@ -535,6 +535,29 @@ def notes(id)
...
+  # ==== Returns
+  # * <~Pagerduty::Note::Note>
+  #   * 'id'<~String>
+  #   * 'user'<~Pagerduty::User>
+  #   * 'id'<~String>
+  #   * 'content'<~String>
+  #   * 'created_at'<~String>
+  # {Pagerduty API Reference}[https://developer.pagerduty.com/documentation/rest/incidents/notes/create]
+  def create_note(options={})
+    Pagerduty::Notes::Note.new(curl({
```
Comment (kryptek) [inline lib/pagerduty/pagerduty.rb]: This should just be Note.new

**Pull request closed without merging**